scripts/k3s-versions.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# This script expects to be called from the root of the repo.
+# It will rebuild resources/k3s-versions.json from both the k3s update
+# channel and the GitHub k3s releases list.
+# Creates a pull request if the new version is different.
+
+set -eu
+
+K3S_VERSIONS="resources/k3s-versions.json"
+
+CHANNELS=$(jq '[.data.[] | select(.name | test("^(stable|latest|v1\\.(2[1-9]|[3-9][0-9]+))$"))] | reduce .[] as $ch ({}; .[$ch.name] = ($ch.latest | scan("^[^+]+") | ltrimstr("v")))' <(curl -s https://update.k3s.io/v1-release/channels))
+
+VERSIONS=$(gh api /repos/k3s-io/k3s/releases --paginate --jq '.[] | select((.draft|not) and (.prerelease|not)) | .tag_name')
+VERSIONS=$(grep -E '^v1\.(2[1-9]|[3-9][0-9]+)\.[0-9]+\+k3s[0-9]+$' <<<"$VERSIONS" | sort -V -r | sort -V -u -k1,1 -t+ | jq --null-input --raw-input '[inputs]')
+
+echo '{"cacheVersion": 2}' | jq --sort-keys --argjson channels "$CHANNELS" --argjson versions "$VERSIONS" '.channels = $channels | .versions = $versions' >"$K3S_VERSIONS"
+
+# Exit if there are no changes
+git diff --exit-code && exit
+
+BRANCH_NAME="k3s-versions-update-$(date +%Y%m%d%H%M%S)"
```

Comment:
This logic will create a new PR every time the workflow is run. It should check if a PR/branch already exists, and if there is a change relative to that branch, just add a commit to it instead of creating another one.